### PR TITLE
Dj/271 maintain one database connection

### PR DESCRIPTION
**Pull request by drujensen**
This fixes the connection issue #271

### CODE REVIEW

File: spec/adapter/adapters_spec.cr [hunk 1]
```diff
@@ -16,7 +16,7 @@ describe Granite::Adapters do
 
     it "should disallow multiple adapters with the same name" do
       expect_raises(Exception, "Adapter with name 'mysql' has already been registered.") do
-        Granite::Adapters << Granite::Adapter::Pg.new({name: "mysql", url: "PG_DATABASE_URL"})
+        Granite::Adapters << Granite::Adapter::Pg.new({name: "mysql", url: ENV["PG_DATABASE_URL"]})
```
Comment (robacarp) [inline spec/adapter/adapters_spec.cr]: this probably doesn't matter, the test is expecting to catch an exception before the url is ever acted on

File: src/adapter/base.cr [hunk 2]
```diff
@@ -7,14 +7,16 @@ require "db"
...
 
   def initialize(connection_hash : NamedTuple(url: String, name: String))
     @name = connection_hash[:name]
     @url = connection_hash[:url]
+    @database = DB.open(@url)
   end
 
   def open(&block)
-    yield DB.open(@url)
+    yield database
```
Comment (robacarp) [inline src/adapter/base.cr]: 💯

**Comment by vlazar** [comment]
@robacarp FYI I've submitted PR to TFB to update Crystal and frameworks to the latest version https://github.com/TechEmpower/FrameworkBenchmarks/pull/4020

The tests are passing now, but I think I should maybe wait for this fix in Granite ORM for better benchmarks results for Amber?

Let me know if there is a bugfix release for Granite ORM planned soon, so I can update my PR before it gets merged in.

**Comment by drujensen** [comment]
@vlazar I will release this today.

**Pull request merged**